packages/valory/skills/decision_maker_abci/behaviours/storage_manager.py

```diff
@@ -253,19 +270,102 @@ def _fetch_accuracy_info(self) -> Generator[None, None, bool]:
                 f"Could not retrieve data from the url {accuracy_link}. "
                 f"Received status code {response.status_code}."
             )
-            return False
+            return None
 
-        self.context.logger.info("Parsing accuracy information of the tools...")
         try:
-            self.accuracy_information = StringIO(response.body.decode())
+            tool_accuracy_data = StringIO(response.body.decode())
+            self.context.logger.info("Parsing accuracy information of the tools...")
+            return tool_accuracy_data
+
         except (ValueError, TypeError) as e:
             self.context.logger.error(
                 f"Could not parse response from ipfs server, "
                 f"the following error was encountered {type(e).__name__}: {e}"
             )
-            return False
+            return None
 
-        return True
+    def _set_accuracy_info_from_remote(self) -> Generator[None, None, None]:
+        """Set the tool accuracy info from the remote store."""
+        # Fetch accuracy information from remote
+        tool_accuracy_data = yield from self._fetch_accuracy_info()
+        if tool_accuracy_data:
+            self.accuracy_information = tool_accuracy_data
+
+    def _read_local_policy_date(self) -> int:
+        """Reads the updated timestamp from the policy file."""
+
+        self.context.logger.info("Reading policy updated timestamp...")
+        policy = self._read_tool_accuracy_local()
+        if policy:
+            policy_json = json.loads(policy)
+            updated_timestamp = policy_json.get("updated_ts")
+
+            if updated_timestamp is str:
+                self.context.logger.info(
+                    f"Local updated timestamp found {updated_timestamp}"
+                )
+                return updated_timestamp
+            else:
+                self.context.logger.info("No timestamp found. Using minium: 0")
+                return 0
+        return 0
+
+    def _fetch_remote_tool_date(self) -> Generator[None, None, int]:
+        """Fetch the max transaction date from the remote accuracy storage."""
+        self.context.logger.info("Checking remote accuracy information date... ")
+        self.context.logger.info("Trying to read max date in file...")
+        tool_accuracy_data = yield from self._fetch_accuracy_info()
```

Comment:
Why fetching in two different methods?

Reply:
I'm not sure I understand. The `_fetch_accuracy_info` always fetches from the remote.